src/main/java/seedu/address/logic/parser/ListAppointmentCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENT_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.ListAppointmentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.StudentIdContainsIndexPredicate;
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListAppointmentCommandParser implements Parser<ListAppointmentCommand> {
+    @Override
+    public ListAppointmentCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_STUDENT_ID);
+
+        List<Predicate<Appointment>> predicates = new ArrayList<>();
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_STUDENT_ID);
+
+        // All these criterias are OR not AND
+//        if (argMultimap.getValue(CliSyntax.PREFIX_NAME).isPresent()) {
+//            Name name = ParserUtil.parseName(argMultimap.getValue(CliSyntax.PREFIX_NAME).get());
+////            predicates.add(new NameContainsKeywordsPredicate(Collections.singletonList(name.fullName)));
+//        }
+
+        if (argMultimap.getValue(PREFIX_STUDENT_ID).isPresent()) {
+            int studentId = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_STUDENT_ID).get()).getOneBased();
+            predicates.add(new StudentIdContainsIndexPredicate(studentId));
+        }
+
+        // Combine predicates with AND logic
+        Predicate<Appointment> combinedPredicate = predicates.stream()
+                .reduce(p -> true, Predicate::and);
+
+        return new ListAppointmentCommand(combinedPredicate);
+    }
+
+    /**
+     * Returns true if any of the prefixes contains non-empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean hasAtLeastOnePrefixPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).anyMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```

Comment:
I actually think this method and other commands like `arePrefixesPresent()` can be instead moved into the ParserUtil

Reply:
Hmm, I do agree with you. Will make the appropriate changes.